app/models/stock.rb

```diff
@@ -1,5 +1,13 @@
 class Stock < ActiveRecord::Base
   has_many :orders
+  has_many :buy_orders
+  has_many :sell_orders
+
+  scope :for_user, -> (user) { includes(:orders).where(orders: {user: user}) }
 
   validates :name, :symbol, presence: true
+
+  def market_price
```

Comment:
I really should have put this comment by this code:

```ruby
+  # Not really sure that it's appropriate to be ordering on this scope
+  scope :fulfilled, -> { where.not(fulfilled_at: nil).order(fulfilled_at: :desc) }
```

I think it would indeed be better to split `.order(fulfilled_at: :desc)` into a separate scope.